src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1178,6 +1178,11 @@ public function originalIsEquivalent($key, $current)
         } elseif ($this->hasCast($key, ['object', 'collection'])) {
             return $this->castAttribute($key, $current) ==
                 $this->castAttribute($key, $original);
+        } elseif ($this->hasCast($key, 'float')) {
+            return bccomp(
```

Comment:
I think by using `bccomp` function, we need to require `ext-bcmath` into the `composer.json` file

Reply:
This project has a depency on: `moontoast/math`.

So therefore the existence of `ext-bcmath` is already checked: https://github.com/moontoast/math/blob/master/composer.json#L14

Reply:
@BassemN is correct though; that _another_ dependency depends on something which is now ultimately a shared dependency is not enough.

> This project has a depency on: moontoast/math.

It's only a _suggested_ dependency of illuminate/support.

However you can use illuminate/database independent of Laravel (which I'm doing).

Therefore I believe that the dependency belongs into:
- `src/Illuminate/Database/composer.json`
  as well as into 
- `composer.json`

Because it's now a hard requirement for Laravel 6.

Reply:
We cannot add that dependency. I'd suggest we change this code to avoid using that function.

Reply:
lol :smile: